components/canvas/canvas_paint_thread.rs

```diff
@@ -558,20 +558,35 @@ impl<'a> CanvasPaintThread<'a> {
     fn send_data(&mut self, chan: IpcSender<CanvasData>) {
         self.drawtarget.snapshot().get_data_surface().with_data(|element| {
             let size = self.drawtarget.get_size();
-            self.webrender_api.update_image(self.webrender_image_key,
-                                            webrender_traits::ImageDescriptor {
-                                                width: size.width as u32,
-                                                height: size.height as u32,
-                                                stride: None,
-                                                format: webrender_traits::ImageFormat::RGBA8,
-                                                offset: 0,
-                                                is_opaque: false,
-                                            },
-                                            element.into(),
-                                            None);
+
+            let descriptor = webrender_traits::ImageDescriptor {
+                                width: size.width as u32,
```

Comment:
nit: Indentation looks weird here. Either indent membersfour spaces from `webrender_traits`, or move it to the left so the members are aligned with `descriptor`